app_dart/lib/src/foundation/utils.dart

```diff
@@ -124,17 +131,33 @@ Future<List<Target>> getTargetsToRun(Iterable<Target> targets, List<String?> fil
     final List<String> globs = target.value.runIf;
     // Handle case where [Target] initializes empty runif
     if (globs.isEmpty) {
-      targetsToRun.add(target);
-    }
-
-    for (String glob in globs) {
-      glob = glob.replaceAll('**', '[A-Za-z0-9_/.]+');
-      glob = glob.replaceAll('*', '[A-Za-z0-9_.]+');
-      // If a file is found within a pre-set dir, the builder needs to run. No need to check further.
-      final RegExp regExp = RegExp('^$glob\$');
-      if (glob.isEmpty || files.any((String? file) => regExp.hasMatch(file!))) {
+      // Evaluate run_if_not.
+      final List<String> negativeGlobs = target.value.runIfNot;
```

Comment:
Rather than getting the value could you not just create a runIfNot getter?

Reply:
From the existing code Target was designed as a wrapper to the proto class potentially to avoid having to keep the wrapper and the proto in sync.